animation_nodes/math/rotation_conversion.pyx

```diff
@@ -122,3 +127,119 @@ cdef void normalizedAxisCosAngleToMatrix(Matrix3_or_Matrix4* m, Vector3* axis, f
     if Matrix3_or_Matrix4 is Matrix4:
         m.a14, m.a24, m.a34 = 0, 0, 0
         m.a41, m.a42, m.a43, m.a44 = 0, 0, 0, 1
+
+# Euler to Quaternion
+##########################################################    
+        
+# https://en.wikipedia.org/wiki/Conversion_between_quaternions_and_Euler_angles
+cdef euler3ToQuaternion(Quaternion* q, Euler3 *e):
+    cdef float cx = cos(e.x * 0.5)
+    cdef float sx = sin(e.x * 0.5)
+    cdef float cy = cos(e.y * 0.5)
+    cdef float sy = sin(e.y * 0.5)
+    cdef float cz = cos(e.z * 0.5)
+    cdef float sz = sin(e.z * 0.5)
+    
+    cdef float cxcy = cx * cy
+    cdef float sxcy = sx * cy
+    cdef float sxsy = sx * sy
+    cdef float cxsy = cx * sy
+    
+    q.w = cxcy * cz + sxsy * sz
+    q.x = sxcy * cz - cxsy * sz
+    q.y = cxsy * cz + sxcy * sz
+    q.z = cxcy * sz - sxsy * cz
+    
+# Quaternion to Euler
+##########################################################
+
+# https://en.wikipedia.org/wiki/Conversion_between_quaternions_and_Euler_angles
+cdef quaternionToEuler3(Euler3 *e, Quaternion *q):
+    quaternionNormalize_InPlace(q)
+    cdef float sinrCosp = 2 * (q.w * q.x + q.y * q.z)
+    cdef float cosrCosp = 1 - 2 * (q.x * q.x + q.y * q.y)
+    e.x = atan2(sinrCosp, cosrCosp)
+
+    cdef float sinp = 2 * (q.w * q.y - q.z * q.x)
+    if abs(sinp) >= 1.0:
+        e.y = copysign(PI / 2, sinp)
+    else:
+        e.y = asin(sinp)
+
+    cdef float sinyCosp = 2 * (q.w * q.z + q.x * q.y)
+    cdef float cosyCosp = 1 - 2 * (q.y * q.y + q.z * q.z)
+    e.z = atan2(sinyCosp, cosyCosp)
+    e.order = 0
+
+# Quaternion to Matrix
+##########################################################
+
+# https://www.euclideanspace.com/maths/geometry/rotations/conversions/quaternionToMatrix/index.htm
+cdef quaternionToMatrix4(Matrix4 *m, Quaternion *q):
+    quaternionNormalize_InPlace(q)
+    cdef float w, x, y, z
+    w, x, y, z = q.w, q.x, q.y, q.z
+    cdef float ww = w * w
+    cdef float xx = x * x
+    cdef float yy = y * y
+    cdef float zz = z * z
+
+    cdef invs = 1 / (xx + yy + zz + ww)
```

Comment:
Put `@cython.cdivision(True)` at the top of the function when the function have divisions or modulo. If the division can be a zero division, put an explicit check for zero.

Reply:
@kdsin23 This wasn't handled.